roles/matrix-nginx-proxy/tasks/ssl/setup_ssl_lets_encrypt.yml

```diff
@@ -17,13 +17,71 @@
 #
 
 - block:
+    - set_fact:
+        matrix_ssl_lets_encrypt_certbot_docker_image: "{{ matrix_ssl_lets_encrypt_certbot_custom_docker_image if matrix_ssl_lets_encrypt_certbot_challenge_image == 'custom' else matrix_ssl_lets_encrypt_certbot_dns_docker_image if matrix_ssl_lets_encrypt_certbot_challenge_image == 'dns' else matrix_ssl_lets_encrypt_certbot_http_docker_image }}"
```

Comment:
Wny not define this variable where it was (in `defaults/main.yml`), instead of only having it available at runtime?

You can do the same for `matrix_ssl_lets_encrypt_certbot_docker_image_force_pull`.

Reply:
I moved it at runtime because I though it could bring confusion as you are not supposed to set this variable through configuration.
You need to rely on `matrix_ssl_lets_encrypt_certbot_challenge_image` instead.
One of the drawback of this is that I had to also move `matrix_ssl_lets_encrypt_certbot_docker_image_force_pull` at runtime either which makes less sense.

No problem to move it back in `defaults/main.yml`.